cli/cmd/encore/deploy.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/cockroachdb/errors"
+	"github.com/logrusorgru/aurora/v3"
+	"github.com/spf13/cobra"
+
+	"encr.dev/cli/cmd/encore/cmdutil"
+	"encr.dev/cli/internal/platform"
+	"encr.dev/pkg/appfile"
+)
+
+var (
+	appSlug string
+	envName string
+	commit  string
+	branch  string
+	format  string
+)
+
+var deployAppCmd = &cobra.Command{
+	Use:                   "deploy --commit COMMIT_SHA | --branch BRANCH_NAME",
+	Short:                 "Deploy an Encore app to a cloud environment",
+	DisableFlagsInUseLine: true,
+	Hidden:                true,
+	Run: func(c *cobra.Command, args []string) {
+		if commit != "" {
+			hb, err := hex.DecodeString(commit)
+			if err != nil || len(hb) != 20 {
+				cmdutil.Fatalf("invalid commit: %s", commit)
+			}
+		}
+		if appSlug == "" {
+			appRoot, _, err := cmdutil.MaybeAppRoot()
+			if err != nil {
+				cmdutil.Fatalf("no app found. Run deploy inside an encore app directory or specify the app with --app")
+			}
+			appSlug, err = appfile.Slug(appRoot)
+			if err != nil {
+				cmdutil.Fatalf("no app found. Run deploy inside an encore app directory or specify the app with --app")
+			}
+		}
+		if envName == "" {
+			envName = "@primary"
```

Comment:
Is this a good idea? Feels a bit risky to not be explicit about which environment to deploy to.

Reply:
Yeah, you're probably right. Was striving to be consistent with other commands.